utils/http_utils.go

```diff
@@ -22,52 +23,59 @@ type HttpResponse struct {
 	StatusCode int
 }
 
-// It send an http post to the endpoint signing the body with the worker's private key
-func SignedPost(workerId string, body interface{}, endpoint string) *HttpResponse{
-	requestBody, err := json.Marshal(body)
+func getSignature(payload interface{}, workerId string) []byte {
+	parsedPayload, err := json.Marshal(payload)
 
 	if err != nil {
-		log.Fatal("Error on marshalling the request body")
+		log.Fatal("Error on marshalling the payload")
 	}
 
-	data, hashSum := SignMessage(GetPrivateKey(workerId), requestBody)
+	signedPayload, _ := SignMessage(GetPrivateKey(workerId), parsedPayload)
 
-	payload := &map[string][]byte{"data": data, "hashSum": hashSum}
+	return signedPayload
+}
 
-	return Post(payload, endpoint)
+func AddSignature(workerId string, payload interface{}, headers http.Header) http.Header {
+	signature := GetSignature(payload, workerId)
+	headers.Set("SIGNATURE", string(signature))
+	return headers
```

Comment:
I believe that it is not necessary to return an http.Header because you already modify the headers parameter.

Reply:
We've discussed about it before and I agreed with you previously, but now, I guess this is the best approach, as much as we can avoid state is better, don't you agree?